hrm-domain/scheduled-tasks/case-status-transition/dataAccess.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright (C) 2021-2023 Technology Matters
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see https://www.gnu.org/licenses/.
+ */
+
+import { db } from '@tech-matters/hrm-core/connection-pool';
+import { CaseStatusTransitionRule } from './caseStatusTransitionRule';
+import { AccountSID } from '@tech-matters/types';
+
+/**
+ * Apply the transition rule to any qualifying cases in the DB and returns the IDs of the cases that were updated
+ * @param accountSid
+ * @param rule
+ * @returns void
+ */
+export const applyTransitionRuleToCases = async (
+  accountSid: AccountSID,
+  rule: CaseStatusTransitionRule,
+): Promise<string[]> => {
+  const records = await db.task(async conn => {
+    return conn.manyOrNone<{ id: string }>(
+      `UPDATE "Cases" 
+                SET 
+                "status" = $<targetStatus>, "statusUpdatedAt" = CURRENT_TIMESTAMP, 
+                "statusUpdatedBy" = $<description> 
```

Comment:
It's not super intuitive to me to assign `description` to `statusUpdatedBy`. Can we leave a comment here describing the use case?